tests/setData.spec.ts

```diff
@@ -214,4 +214,35 @@ describe('setData', () => {
     expect(wrapper.vm.value).toBeInstanceOf(Date)
     expect(wrapper.vm.value!.toISOString()).toBe('2022-08-11T12:15:54.000Z')
   })
+
+  it('should retain prototype methods for constructed objects when calling setData', async () => {
+    const expectedResult = 'success!'
+    class TestClass {
+      getResult() {
+        return expectedResult
+      }
+    }
+
+    const wrapper = mount(
+      defineComponent({
+        template: '<div/>',
+        data() {
+          return { value: new TestClass() }
```

Comment:
Maybe the initial value should be something else to see if `setData` properly updates it no?

Reply:
As in a different class altogether? 
Or just with some properties on the class that get changed?

Reply:
for example:
```
it('should retain prototype methods for constructed objects when calling setData', async () => {
  const expectedResult = 'success!'
  class TestClass {
    name: string
    constructor(name: string) {
      this.name = name
    }
    getResult(): string {
      return expectedResult
    }
  }

  const wrapper = mount(
    defineComponent({
      template: '<div/>',
      data() {
        return { value: new TestClass('test1') }
      },
      methods: {
        getResult() {
          return `${this.value.name}: ${this.value.getResult()}`
        }
      }
    })
  )

  expect(wrapper.vm.getResult()).toStrictEqual(`test1: ${expectedResult}`)

  await wrapper.setData({
    value: new TestClass('test2')
  })

  expect(wrapper.vm.getResult()).toStrictEqual(`test2: ${expectedResult}`)
})
```